misc/material-migration/migrate-material.rb

```
@@ -0,0 +1,1537 @@
+require 'json'
+require 'securerandom'
+require 'rest-client'
+require 'dotenv/load'
+require 'pathname'
+require 'kramdown'
+
+class MaterialMigrator
+  def initialize(directory)
+    unless File.directory?(directory)
+      raise ArgumentError, "#{directory} is not a directory."
+      exit
```

RuboCop (1.87.0)

[warning] Unreachable code detected.

(Lint/UnreachableCode)